spring-integration-file/src/main/java/org/springframework/integration/file/FileReadingMessageSource.java

```diff
@@ -161,6 +165,9 @@ public FileReadingMessageSource(int internalQueueCapacity) {
 	public FileReadingMessageSource(Comparator<File> receptionOrderComparator) {
 		this.toBeReceived = new PriorityBlockingQueue<File>(
 				DEFAULT_INTERNAL_QUEUE_CAPACITY, receptionOrderComparator);
+		if (WATCH_SERVICE_PRESENT) {
+			this.watchEvents = new WatchEventType[] { WatchEventType.CREATE };
+		}
```

Comment:
Yeah! Agree about `Assert.state()` in the `setUseWatchService()`, but not sure what that event should mean...